packages/core/src/external/quest/fhir/observation.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  CodeableConcept,
+  Coding,
+  Identifier,
+  Observation,
+  Patient,
+  ObservationReferenceRange,
+} from "@medplum/fhirtypes";
+import { uuidv7 } from "@metriport/shared/util/uuid-v7";
+import { ResponseDetail } from "../schema/response";
+import { getPatientReference } from "./patient";
+import { getQuestDataSourceExtension } from "./shared";
+import { LOINC_URL, CPT_URL } from "@metriport/shared/medical";
+import { HL7_OBSERVATION_INTERPRETATION_SYSTEM, QUEST_LOCAL_RESULT_CODE_SYSTEM } from "./constant";
+
+type ObservationValue = Pick<Observation, "valueQuantity" | "valueString" | "valueCodeableConcept">;
+
+export function getObservation(
+  detail: ResponseDetail,
+  { patient }: { patient: Patient }
+): Observation {
+  const identifier = getObservationIdentifier(detail);
+  const code = getObservationCode(detail);
+  const subject = getPatientReference(patient);
+  const interpretation = getObservationInterpretation(detail);
+  const referenceRange = getObservationReferenceRange(detail);
+  const { valueQuantity, valueString } = getObservationValue(detail);
+  const extension = [getQuestDataSourceExtension()];
+
+  return {
+    resourceType: "Observation",
+    id: uuidv7(),
+    status: "final",
+    subject,
+    ...(valueQuantity ? { valueQuantity } : {}),
+    ...(valueString ? { valueString } : {}),
+    ...(referenceRange ? { referenceRange } : {}),
+    ...(interpretation ? { interpretation } : {}),
+    ...(identifier ? { identifier } : {}),
+    ...(code ? { code } : {}),
+    extension,
+  };
```

Comment:
needs:
- basedOn reference to the corresponding service request resource

Reply:
is there a reason why there is no category? can we not infer one of these values from the data we get from quest?
<img width="800" height="781" alt="image" src="https://github.com/user-attachments/assets/61effd7d-9f53-4abe-9687-4c5e12230e6a" />

Reply:
- resultComments (very last column in source data) can be used for the `note` field on the observation

Reply:
As per our huddle discussion, will keep the notes on the diagnostic report.